lua/examples/policy.lua

```diff
@@ -126,26 +85,39 @@ local function lookup_simple_policy(simple_policy, pkt_info)
 				policylib.c.ICMPV6_ECHO_REQUEST_TYPE and
 				icmpv6_code ==
 				policylib.c.ICMPV6_ECHO_REQUEST_CODE then
-			return declined
+			return dcs_declined
 		end
 
-		return default
-	else
+		return dcs_default
+	end
+
+	local l3_policy = simple_policy[pkt_info.inner_ip_ver]
+	if l3_policy == nil then
 		return nil
 	end
 
-	for i, v in ipairs(simple_policy[pkt_info.inner_ip_ver]) do
-		for j, g in ipairs(v) do
-			if g["dest_port"] == dest_port then
-				return g["policy_id"]
-			end
+	if pkt_info.l4_proto == policylib.c.TCP then
+		if pkt_info.upper_len < ffi.sizeof("struct tcp_hdr") then
+			return dcs_malformed
+		end
+
+		local tcphdr = ffi.cast("struct tcp_hdr *", pkt_info.l4_hdr)
+		return l3_policy[tcphdr.dst_port]
+	end
+
+	if pkt_info.l4_proto == policylib.c.UDP then
+		if pkt_info.upper_len < ffi.sizeof("struct udp_hdr") then
+			return dcs_malformed
 		end
+
+		local udphdr = ffi.cast("struct udp_hdr *", pkt_info.l4_hdr)
```

Comment:
`struct udp_hdr`, `struct tcp_hdr`, and `struct icmp_hdr` and possibly others need `rte_` prepended to the struct name.